resources/sqlite/src/lib.rs

```diff
@@ -0,0 +1,172 @@
+//! Shuttle resource providing a SQLite database. The database will be created in-process by the shuttle runtime.
+//!
+//! ## Example
+//! Simply annotate your main function to get a [`sqlx::SqlitePool`](https://docs.rs/sqlx/latest/sqlx/type.SqlitePool.html)
+//! with default configuration.
+//! Pass it to [`sqlx::query`](https://docs.rs/sqlx/latest/sqlx/macro.query.html) to interact with the database.
+//! ```ignore
+//! #[shuttle_runtime::main]
+//! async fn axum(
+//!     #[shuttle_sqlite::SQLite] pool: shuttle_sqlite::SqlitePool,
+//! ) -> shuttle_axum::ShuttleAxum {
+//!     let _ = sqlx::query(
+//!         "CREATE TABLE IF NOT EXISTS users(id int, name varchar(128), email varchar(128));",
+//!     )
+//!     .execute(&pool)
+//!     .await
+//!     .unwrap();
+//! }
+//! ```
+//!
+//! ## Configuration
+//! The database can be configured using [`SQLiteConnOpts`] which mirrors sqlx's [`SqliteConnectOptions`](https://docs.rs/sqlx/latest/sqlx/sqlite/struct.SqliteConnectOptions.html) for the
+//! options it exposes.
+//!
+//! Construction of the full connection string is handled internally for security reasons and defaults to creating a
+//! file-based database named `default_db.sqlite` with `create_if_missing == true`. Use the `filename` and/or
+//! `in_memory` methods to configure the type of database created.
+//!
+//! See [`SqliteConnectOptions::new()`](https://docs.rs/sqlx/latest/sqlx/sqlite/struct.SqliteConnectOptions.html#method.new)
+//! for all default settings.
+//!
+//! ```ignore
+//! #[shuttle_runtime::main]
+//! async fn axum(
+//!     #[shuttle_sqlite::SQLite(opts = SQLiteConnOpts::new().filename("custom.sqlite"))] pool: shuttle_sqlite::SqlitePool,
+//! ) -> shuttle_axum::ShuttleAxum { /* ... */ }
+//! ```
+//! Note that Shuttle does currently not support the `collation`, `thread_name`, `log_settings`, `pragma`, `extension`,
+//! `shared_cache` options.
+use async_trait::async_trait;
+use serde::Serialize;
+use shuttle_service::{Factory, ResourceBuilder, Type};
+// use tracing::debug;
+
+mod conn_opts;
+pub use conn_opts::*;
+
+use sqlx::sqlite::SqliteConnectOptions;
+/// The [`sqlx::SqlitePool`](https://docs.rs/sqlx/latest/sqlx/type.SqlitePool.html) that is being returned to the user.
+///
+pub use sqlx::SqlitePool;
+
+/// Builder struct used to configure the database, e.g. `SQLite(opts = SQLiteConnOpts::new())`.
+#[derive(Serialize)]
+pub struct SQLite {
+    opts: SQLiteConnOpts,
+}
```

Comment:
Right. I liked this because it didn't collide with the original structs like `sqlx::Sqlite`. But I agree, naming conventions should be followed, will rename to something like `ShuttleSqlite`.